### PR TITLE
feat: add bundle analyzer in web builder

### PR DESCRIPTION
**Pull request by sanyuan0704**
# PR Details

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/builder/web-builder/src/plugins/bundleAnalyzer.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        const { BundleAnalyzerPlugin } = await import(
+          'webpack-bundle-analyzer'
+        );
+        chain
+          .plugin(CHAIN_ID.PLUGIN.BUNDLE_ANALYZER)
+          .use(BundleAnalyzerPlugin, [
+            {
+              analyzerMode: 'static',
+              openAnalyzer: false,
+              reportFilename: 'report.html',
```
Comment (chenjiahan) [inline packages/builder/web-builder/src/plugins/bundleAnalyzer.ts:21]: reportFilename 的默认值可以根据 modifyWebpackChain 第二个参数里的 target 做一下区分，比如 `report-web.html` `report-node.html`，因为 SSR 构建时会有多个 target，如果同名会导致覆盖

File: packages/builder/web-builder/package.json [hunk 2]
```diff
@@ -58,14 +58,16 @@
     "style-loader": "3.3.1",
     "terser-webpack-plugin": "5.3.3",
     "ts-loader": "9.3.1",
-    "webpack": "^5.71.0"
+    "webpack": "^5.71.0",
+    "webpack-bundle-analyzer": "4.5.0"
```
Comment (chenjiahan) [inline packages/builder/web-builder/package.json:62]: 预打包也加一下~

File: packages/builder/web-builder/tests/plugins/bundleAnalyzer.test.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { expect, describe, it } from 'vitest';
+import { PluginBabel } from '../../src/plugins/babel';
+import { createStubBuilder } from '../utils/builder';
+
+describe('plugins/bundleAnalyze', () => {
+  it('should add bundle analyze plugin', async () => {
+    const builder = createStubBuilder({
+      plugins: [PluginBabel()],
```
Comment (chenjiahan) [inline packages/builder/web-builder/tests/plugins/bundleAnalyzer.test.ts:8]: 这里是不是引用错啦，PluginBabel -> PluginBundleAnalyze

File: packages/builder/web-builder/package.json [hunk 4]
```diff
@@ -66,6 +66,7 @@
     "@types/babel__core": "^7.1.19",
     "@types/babel__preset-env": "^7.9.2",
     "@types/node": "^14",
+    "@types/webpack-bundle-analyzer": "4.4.1",
```
Comment (chenjiahan) [inline packages/builder/web-builder/package.json:69]: types 可以删除咯

File: packages/builder/web-builder/tests/plugins/bundleAnalyzer.test.ts (hunk 3 above)
Comment (sanyuan0704) [reply packages/builder/web-builder/tests/plugins/bundleAnalyzer.test.ts:8]: 对，插件引错了，我重新生成下 snapshot

**Review by chenjiahan — APPROVED** [review]

File: packages/builder/web-builder/src/types/config/performance.ts [hunk 5]
```diff
@@ -1,6 +1,9 @@
+import type { BundleAnalyzerPlugin } from 'webpack-bundle-analyzer';
```
Comment (chenjiahan) [inline packages/builder/web-builder/src/types/config/performance.ts:1]: 这里要改成 compiled 里的 path 哈

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**